### PR TITLE
Handle DeletedFinalStateUnknown events form cache

### PR DESCRIPTION
**Pull request by c-kruse**
Closes https://github.com/SumoLogic/sumologic-otel-collector/issues/1277

### CODE REVIEW

File: pkg/processor/k8sprocessor/kube/client.go [hunk 1]
```diff
@@ -193,11 +193,28 @@ func (c *WatchClient) handlePodUpdate(old, new interface{}) {
...
+	switch obj := obj.(type) {
+	case *api_v1.Pod:
+		pod = obj
+	case cache.DeletedFinalStateUnknown:
+		prev, ok := obj.Obj.(*api_v1.Pod)
+		if !ok {
+			c.logger.Error(
+				"object received was DeletedFinalStateUnknown but did not contain api_v1.Pod",
+				zap.Any("received", obj),
+			)
```
Comment (sumo-drosiek) [inline pkg/processor/k8sprocessor/kube/client.go:208]: is it logged often? We would like to avoid extensive error logging if this may be expected behavior

Comment (astencel-sumo) [reply pkg/processor/k8sprocessor/kube/client.go:208]: If I understand correctly, this is more of a sanity check that shouldn't trigger an error in normal situation, as by design the `cache.DeletedFinalStateUnknown` event should contain the object it pertains to.

**Review by astencel-sumo — APPROVED** [review]
Thanks a lot for this fix @c-kruse! :clap:

File: pkg/processor/k8sprocessor/kube/client.go (hunk 1 above)
Comment (sumo-drosiek) [reply pkg/processor/k8sprocessor/kube/client.go:208]: I needed to double check if this function should always get pod object and confirm that as observer is run per object type

**Review by sumo-drosiek — APPROVED** [review]

**Pull request merged**